pkg/provider/provider.go

```diff
@@ -93,7 +96,12 @@ func (p *KedaProvider) GetExternalMetric(namespace string, metricSelector labels
 				if err != nil {
 					logger.Error(err, "error getting metric for scaler", "ScaledObject.Namespace", scaledObject.Namespace, "ScaledObject.Name", scaledObject.Name, "Scaler", scaler)
 				} else {
+					for _, metric := range metrics {
+						metricValue, _ := metric.Value.AsInt64()
+						p.metricServer.RecordHPAScalerMetrics(namespace, metric.MetricName, metricSelector.String(), metricValue)
+					}
 					matchingMetrics = append(matchingMetrics, metrics...)
+					p.metricServer.RecordHPAScalerErrorTotals(namespace, info.Metric, metricSelector.String(), err)
```

Comment:
Shouldn't this be in the `if err != nil {}` block above?

Reply:
I am glad you mentioned this. The reasoning behind it's placement is to handle the initial registration of the error metric - https://github.com/tbickford/keda/blob/edc011bf19378ae39e6269d0a336654365ad9083/pkg/metrics/prometheus_metrics.go#L65 (example - `keda_hpa_scaler_errors_total{metric="queuelength",namespace="default",selector="deploymentName=rabbitmq-keda-consumer-test"} 0`). I am not super happy with this logic. Alternatively I could do try to initially register the metric for each Scaler on startup but then I would also have to handle newly registered ScaledObjects which may not really save much from the performance and coding perspective. What are your thoughts?

Reply:
OK, but still, I might be missing something, but where do you actually record the erros? Currently the `p.metricsExporter.RecordHPAScalerErrorTotals(namespace, info.Metric, metricSelector.String(), err)` call is in the block where `err` is always `nil`

Reply:
Nice catch - that is a mistake. It should be outside the block - I will correct.